aws_lambda_powertools/utilities/data_classes/cloud_watch_alarm_event.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+import json
+from enum import Enum, auto
+from typing import List, Optional
+
+from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
+
+
+class CloudWatchAlarmStateValue(Enum):
+    OK = auto()
+    ALARM = auto()
+    INSUFFICIENT_DATA = auto()
+
+
+class CloudWatchAlarmState(DictWrapper):
+    @property
+    def value(self) -> CloudWatchAlarmStateValue:
+        """
+        Overall state of the alarm.
+        """
+        return CloudWatchAlarmStateValue[self["value"]]
+
+    @property
+    def reason(self) -> Optional[str]:
+        """
+        Reason why alarm was changed to this state.
+        """
+        return self.get("reason")
+
+    @property
+    def reason_data(self) -> Optional[dict]:
+        """
+        Additional data to back up the reason, usually contains the evaluated data points,
+        the calculated threshold and timestamps.
+        """
+        if self.get("reasonData") is None:
+            return None
+
+        return json.loads(str(self.get("reasonData")))
```

Comment:
We have 2 things to do here:

1 - It is good to have a field that returns the raw value, that is, customers may not want to deserialize the value and want to use the raw string. So keep this field just returning `self.get("reasonData")` and create a new one called `reason_data_decoded`
2 - In this new field, allow the customer to bring their own json deserializer, as we do here https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/data_classes/active_mq_event.py#L28